client/my-sites/sidebar-unified/collapse-sidebar.jsx

```diff
@@ -0,0 +1,70 @@
+/**
+ * Collapse Sidebar Menu Item
+ *
+ **/
+
+/**
+ * External dependencies
+ */
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { isWithinBreakpoint } from '@automattic/viewport';
+
+/**
+ * Internal dependencies
+ */
+import SidebarItem from 'calypso/layout/sidebar/item';
+import SidebarCustomIcon from 'calypso/layout/sidebar/custom-icon';
+import { getSidebarIsCollapsed } from 'calypso/state/ui/selectors';
+import { collapseSidebar, expandSidebar } from 'calypso/state/ui/actions';
+
+export const CollapseSidebar = ( {
+	title,
+	icon,
+	sidebarIsCollapsed,
+	collapseTheSidebar,
+	expandTheSidebar,
+} ) => {
+	const collapsed = sidebarIsCollapsed && isWithinBreakpoint( '>800px' );
+
+	useEffect( () => {
```

Comment:
As we're mutating the DOM here in a way which will change the appearance of the entire page we might need to consider `useLayoutEffect` as that will run before the browser has a chance to paint the DOM updates. Leaving as we have it will allow the browser to paint the component, then the effect will trigger and update the DOM changing the appearance.

Reply:
Great suggestion! Fixed https://github.com/Automattic/wp-calypso/pull/46045/commits/54126f462af6d30edc73d9161e4090cdd5b396bb